src/Microsoft.VisualStudio.Threading/AwaitExtensions.cs

```diff
@@ -157,5 +168,84 @@ public void GetResult()
+        /// </summary>
+        public struct ConfiguredTaskYieldAwaitable
+        {
+            /// <summary>
+            /// A value indicating whether the continuation should run on the captured <see cref="SynchronizationContext"/>, if any.
+            /// </summary>
+            private readonly bool continueOnCapturedContext;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="ConfiguredTaskYieldAwaitable"/> struct.
+            /// </summary>
+            /// <param name="continueOnCapturedContext">A value indicating whether the continuation should run on the captured <see cref="SynchronizationContext"/>, if any.</param>
+            public ConfiguredTaskYieldAwaitable(bool continueOnCapturedContext)
+            {
+                this.continueOnCapturedContext = continueOnCapturedContext;
+            }
+
+            /// <summary>
+            /// Gets the awaiter.
+            /// </summary>
+            /// <returns>The awaiter.</returns>
+            public ConfiguredTaskYieldAwaiter GetAwaiter() => new ConfiguredTaskYieldAwaiter(this.continueOnCapturedContext);
+        }
+
+        /// <summary>
+        /// An awaiter that will always lead the calling async method to yield,
+        /// then immediately resume, possibly on the original <see cref="SynchronizationContext"/>.
+        /// </summary>
+        public struct ConfiguredTaskYieldAwaiter : INotifyCompletion
+        {
+            /// <summary>
+            /// A value indicating whether the continuation should run on the captured <see cref="SynchronizationContext"/>, if any.
+            /// </summary>
+            private readonly bool continueOnCapturedContext;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="ConfiguredTaskYieldAwaiter"/> struct.
+            /// </summary>
+            /// <param name="continueOnCapturedContext">A value indicating whether the continuation should run on the captured <see cref="SynchronizationContext"/>, if any.</param>
+            public ConfiguredTaskYieldAwaiter(bool continueOnCapturedContext)
+            {
+                this.continueOnCapturedContext = continueOnCapturedContext;
+            }
+
+            /// <summary>
+            /// Gets a value indicating whether the caller should yield.
+            /// </summary>
+            /// <value>Always false.</value>
+            public bool IsCompleted => false;
+
+            /// <summary>
+            /// Schedules a continuation to execute immediately (but not synchronously).
+            /// </summary>
+            /// <param name="continuation">The delegate to invoke.</param>
+            public void OnCompleted(Action continuation)
+            {
+                SynchronizationContext syncContext;
+                if (this.continueOnCapturedContext && (syncContext = SynchronizationContext.Current) != null)
```

Comment:
Using '=' in if condition is confusing.

Reply:
Fixed in a newer revision. Although it is a pattern that can help write efficient code without being significantly more verbose.